llama-cli/src/cli_args.rs

```diff
@@ -1,34 +1,94 @@
 use std::path::PathBuf;
 
-use clap::Parser;
-use llama_rs::TokenBias;
-use once_cell::sync::Lazy;
+use clap::{Parser, ValueEnum};
+use llama_rs::{
+    InferenceParameters, InferenceSessionParameters, ModelKVMemoryType, TokenBias, EOT_TOKEN_ID,
+};
+use rand::SeedableRng;
 
 #[derive(Parser, Debug)]
 #[command(author, version, about, long_about = None)]
-pub struct Args {
-    /// Where to load the model path from
-    #[arg(long, short = 'm')]
-    pub model_path: String,
+pub enum Args {
+    #[command()]
+    /// Use a model to infer the next tokens in a sequence, and exit
+    Infer(Box<Infer>),
+
+    #[command()]
+    /// Use a model to dump the tokens for a given prompt
+    DumpTokens(Box<DumpTokens>),
+
+    #[command()]
+    /// Use a model to interactively generate tokens
+    Repl(Box<Repl>),
+
+    #[command()]
+    /// Use a model to interactively generate tokens, and chat with it
+    ///
+    /// Note that most, if not all, existing models are not trained for this
+    /// and do not support a long enough context window to be able to
+    /// have an extended conversation.
+    ChatExperimental(Box<Repl>),
 
-    /// The prompt to feed the generator
+    #[command(hide = true)]
+    /// Convert a PyTorch model to a GGML model
```

Comment:
I would document that this does not currently work. I know it's hidden, but I mean that adding a TODO telling developers that this feature is disabled, and maybe even link at this PR sounds worthwile.